### DISCUSSION TITLE
KeyLike to "alg"

### DISCUSSION THREAD

**Discussion by ekosz** [panva/jose#224] [Ideas]
Hi again,

I was wondering if `jose` would be open to a PR that updates how `SignJWT` works. [Currently](https://github.com/panva/jose/blob/14f7d822811d6097629218159b333f211e1c834b/src/jws/flattened/sign.ts#L116-L118), if the "alg" header is missing during  the signing, `jose` throws an error.  But, for certain `KeyLike`'s (namely, `KeyObject` and `CryptoKey`) we should be able to to do something like the opposite of what's happening in [`generate.ts`](https://github.com/panva/jose/blob/14f7d822811d6097629218159b333f211e1c834b/src/runtime/node/generate.ts#L43-L113) to figure out what the "alg" header should set to given a private key.

Would this project be open to a PR that either:
1. When the "alg" header is missing during "sign", the library first tries to deduce what it should be. And only throw an error if it couldn't figure it out itself.
2. Expose a utility function that takes in a `KeyLike` and returns `?string` of the correct "alg" header

**Use case**: I'm writing an internal library for my company right now for my company using `jose`. The current API tries to hide all the details of the signing and decoding of JWTs. It current takes in an array of `KeyObject`s as its only param. This works for creating of the JWK set, as `fromKeyLike` already is able to take in a `KeyLike` and set the "kty" header. But the same doesn't happen for signing and "alg". I think this would be a nice addition to `jose` and require less code to be written in userland.

**Comment by panva** [comment]
Hi @ekosz 

This is very similar to #172

> Would this project be open to a PR that either:

1. No, I intentionally dropped such a feature when moving from v2.x to v3.x.
2. No, because it is not possible to do so for all KeyLike.

**Reply by ekosz** [reply]
Ah, got it. Sorry I missed discussion. Makes sense!
